### PR TITLE
[Perf]perf: skip unused Cosmos3 I2V conditioning latents

### PR DESCRIPTION
**Pull request by bjf-frz**
PLEASE FILL IN THE PR DESCRIPTION HERE.

## Purpose

  Optimize Cosmos3 I2V latent preparation by avoiding unnecessary VAE encode work for repeated conditioning frames.

  Previously, I2V expanded the input image to a full-length repeated video, encoded the entire video with the VAE, and then only used `cond_latent[:, :, 0:1]`. The remaining conditioning latent frames were masked out and replaced by noise.

  This PR changes I2V to encode only the first conditioning frame as a one-frame video, then writes that `image_latent` into `latents[:, :, 0:1]`. The final latent shape and masking semantics remain unchanged:

  - frame 0: conditioned by the input image
  - frame 1..T: initialized from noise
  - `velocity_mask`: frame 0 fixed, remaining frames denoised

  The VAE latent normalization logic is also factored into a shared helper to keep single-frame and video encode paths consistent.

## Test Plan

  Benchmark setup:

  Model: nvidia/Cosmos3-Nano
  Task: I2V
  Resolution: 1280x720
  Frames: 189
  Steps: 35
  FPS: 24
  Guidance scale: 6.0
  Flow shift: 10.0
  Max sequence length: 4096
  Seeds: 1111, 1112
  GPUs: 4
  Parallel config: cfg2 + usp2 + vae_patch_parallel4 + hsdp
  Concurrency: 1
  Requests: 2 measured + 1 warmup

**vLLM Version:**

**vLLM-Omni Commit:**

## Test Result

  | Version | Completed | E2E latency mean | Throughput | Peak memory |
  |---|---:|---:|---:|---:|
  | baseline | 2/2 | 255375.06 ms | 0.0039 qps | 20318 MB |
  | optimized | 2/2 | 248521.49 ms | 0.0040 qps | 20272 MB |

  E2E latency improvement: 2.68%

### Acc:
Before:

https://github.com/user-attachments/assets/584079a2-a1fc-479b-b92e-e61aeefdcfd7

After:

https://github.com/user-attachments/assets/8a2ca9b8-d0c7-46d0-bc4c-cf874b1ae5dc

**BEFORE SUBMITTING:** read [CONTRIBUTING.md](https://github.com/vllm-project/vllm-omni/blob/main/CONTRIBUTING.md) and run the [precheck-pr skill](https://github.com/vllm-project/vllm-omni/blob/main/.claude/skills/precheck-pr/SKILL.md) with the code agent for a self-check against project conventions.
(anything written below this line will be removed by GitHub Actions)

### CODE REVIEW

File: vllm_omni/diffusion/models/cosmos3/pipeline_cosmos3.py [hunk 1]
```diff
@@ -1786,7 +1764,19 @@ def _encode_conditioning_video(
             scaling_factor = getattr(self.vae.config, "scaling_factor", 1.0)
             latent = latent * scaling_factor
 
-        return latent.to(self.dtype)
+        return latent
+
+    def _encode_conditioning_image_latent(self, image_tensor: torch.Tensor) -> torch.Tensor:
+        """VAE-encode the first I2V conditioning frame.
+
+        I2V only consumes the first conditioning latent frame.  Encoding the
```
Comment (david6666666) [inline vllm_omni/diffusion/models/cosmos3/pipeline_cosmos3.py:1787]: This changes the conditioned latent itself, not just the unused repeated latents. The old path intentionally encoded a repeated video because the temporal VAE can produce degenerate features from a single frame, so please add evidence that the first latent from one-frame encode matches the repeated-video path closely enough, or keep the old conditioning encode semantics.

File: vllm_omni/diffusion/models/cosmos3/pipeline_cosmos3.py [hunk 2]
```diff
@@ -1786,7 +1764,19 @@ def _encode_conditioning_video(
...
+        input image as a one-frame video keeps that latent while avoiding VAE
+        work for repeated frames that are later replaced by noise.
+        """
+        video = image_tensor.unsqueeze(2).to(device=self.device, dtype=self.vae.dtype)
```
Comment (david6666666) [inline vllm_omni/diffusion/models/cosmos3/pipeline_cosmos3.py]: This branch drops the current `_to_vae_device(..., pin_cpu=True)` transfer path from main. When rebasing, please keep that helper here so I2V still uses the pinned/non-blocking H2D copy optimization.

**Review by david6666666 — COMMENTED** [review]
Please address these before merge.

File: vllm_omni/diffusion/models/cosmos3/pipeline_cosmos3.py [hunk 3]
```diff
@@ -1753,29 +1753,7 @@ def _format_and_tokenize_prompts(
...
-        video = image_tensor.unsqueeze(2).expand(-1, -1, num_frames, -1, -1).contiguous()
-        video = video.to(device=self.device, dtype=self.vae.dtype)
-
-        latent = self.vae.encode(video).latent_dist.mode()
-
-        # Normalize (inverse of _decode_latents denormalization)
+    def _normalize_vae_latent(self, latent: torch.Tensor) -> torch.Tensor:
         if hasattr(self.vae.config, "latents_mean") and hasattr(self.vae.config, "latents_std"):
             latents_mean = (
                 torch.tensor(self.vae.config.latents_mean).view(1, -1, 1, 1, 1).to(latent.device, latent.dtype)
```
Comment (david6666666) [inline vllm_omni/diffusion/models/cosmos3/pipeline_cosmos3.py]: Please reuse the cached `_get_latents_mean_std(...)` helper from main when resolving the rebase. Recreating these tensors on each encode path regresses the normalization cache in a perf-sensitive path.

**Comment by david6666666** [comment]
#4340

**Comment by david6666666** [comment]
LGTM

**Review by david6666666 — APPROVED** [review]

**Review by hsliuustc0106 — COMMENTED** [review]
I2V VAE encode optimization looks correct.
